ckanfunctionaltests/api/test_search_datasets.py

```diff
@@ -225,18 +232,20 @@ def test_search_datasets_by_full_slug_specific_field_all_fields_response(
     rsession,
     stable_dataset,
     allfields_term,
+    variables
 ):
-    if allfields_term.startswith("all_fields") and base_url_3.endswith("/3"):
+    if allfields_term.startswith("all_fields") and (base_url_3.endswith("/3") or variables.get('ckan_version') == '2.9'):
         pytest.skip("all_fields parameter not supported in v3 endpoint")
 
-    limit_param, offset_param = _get_limit_offset_params(base_url_3)
+    limit_param, offset_param = _get_limit_offset_params(variables, base_url_3)
 
     response = rsession.get(
         f"{base_url_3}/search/dataset?q=name:{stable_dataset['name']}"
         f"&{allfields_term}&{limit_param}=10"
     )
     assert response.status_code == 200
-    rj = response.json()
+    rj = response.json().get('result') if variables.get('ckan_version') == '2.9' and base_url_3.endswith("/3")\
```

Comment:
...and here.

BUT I've had a go locally at extracting these out into a some helper module, and it didn't seem to help much 😕 The spike commit is here: 19e3e589821e093c16a3519d4d3e37c21090a636

What I'm trying to achieve is next time you need to bump the CKAN version, you only need to update one file, not lots of files. Maybe that's not really practical here.

Reply:
...what I _would_ look at though is the logic on lines 237 and 247 in this file. They look almost identical, but are subtly different - the former has a `endswith "/3" OR CKAN 2.9` and the latter has a `endswith "/3" AND CKAN 2.9`. Is that deliberate or should the logic be the same?

Reply:
This is deliberate, the `all_fields` is unavailable for `\3` or `ckan 2.9` as v1 has been dropped but in order to not update all the things I've returned the json block in the expected way without the `result` root element for v1 endpoints. 

Hope this makes sense, it's a bit weird that CKAN completely removed support for v1/2 as that affects the output but it would be quite a bit of work to add it back in.